helix-term/src/commands/lsp.rs

```diff
@@ -83,6 +83,49 @@ fn lsp_location_to_location(
     })
 }
 
+fn normalize_location_to_utf8(location: &Location, editor: &Editor) -> Location {
+    // Try to get the document text for this URI
+    let maybe_text = if let Some(doc) = editor
+        .documents
+        .values()
+        .find(|doc| doc.uri().as_ref() == Some(&location.uri))
+    {
+        // Document is already open
+        Some(doc.text().clone())
+    } else if let Some(path) = location.uri.as_path() {
+        // Try to open the file temporarily for conversion
+        match std::fs::read_to_string(path) {
+            Ok(content) => Some(Rope::from(content)),
```

Comment:
This is potentially very expensive for I/O and memory - it will read the file into memory once for each location, even if two locations share a URI. I think it's fine to dedup in documents that are already open but we shouldn't be reading potentially many files into memory.